Source/Evolve/WarpXEvolve.cpp

```diff
@@ -541,6 +541,7 @@ void WarpX::SyncCurrentAndRho ()
         SyncRho(rho_fp, rho_cp);
     }
 
+#ifndef WARPX_DIM_RZ
```

Comment:
What is the reason that this is turned off for RZ? The z-boundaries could be PEC.

Reply:
I believe the outer r boundary can also be set to PEC, right? Currently the PEC boundary application does not have any checks that it is only applied in the z-direction when RZ is used. I think it would be good at some point to make the handling of RZ boundaries more consistent with these changes (moving parts of the `ApplyInverseVolumeScalingToChargeDensity` logic to be handled by `ApplyRhofieldBoundary`).

Reply:
Yes, the outer boundary can also be PEC. Note that there will be some trickiness with the outer boundaries because of the r-dependence. I agree about being consistent about handling the axis - it is doing the same thing, copying the charge/current from below the axis to above.

Reply:
So do you think it is okay to merge this as is and then follow-up with the RZ handling later?

Reply:
Maybe the best is then that, for now, I will remove the limitation to non-RZ but add a warning that there will be incorrect handling of the boundary if someone uses a PEC at the upper R boundary.

Reply:
Actually that solution might not work since the test `LaserInjectionFromLASYFile_RZ` uses a PEC boundary for the upper R boundary.